TLM/TLM/Custom/AI/CustomTrainAI.cs

```diff
@@ -715,168 +1014,245 @@ public class CustomTrainAI : TrainAI { // TODO inherit from VehicleAI (in order
             NetManager netMan = Singleton<NetManager>.instance;
-            if ((netMan.m_nodes.m_buffer[(int)transitNodeId].m_flags & NetNode.Flags.TrafficLights) != NetNode.Flags.None) {
-                uint frame = Singleton<SimulationManager>.instance.m_currentFrameIndex;
-                uint simGroup = (uint)transitNodeId >> 7;
-                uint rand = frame - simGroup & 255u;
-                RoadBaseAI.TrafficLightState vehicleLightState;
-                RoadBaseAI.TrafficLightState pedestrianLightState;
-                bool vehicles;
-                bool pedestrians;
-                RoadBaseAI.GetTrafficLightState(transitNodeId, ref netMan.m_segments.m_buffer[(int)position.m_segment], frame - simGroup, out vehicleLightState, out pedestrianLightState, out vehicles, out pedestrians);
-                if (!vehicles && rand >= 196u) {
-                    vehicles = true;
-                    RoadBaseAI.SetTrafficLightState(transitNodeId, ref netMan.m_segments.m_buffer[(int)position.m_segment], frame - simGroup, vehicleLightState, pedestrianLightState, vehicles, pedestrians);
-                }
+            if ((netMan.m_nodes.m_buffer[transitNodeId].m_flags & NetNode.Flags.TrafficLights) == NetNode.Flags.None) {
+                return;
             }
+
+            uint frame = Singleton<SimulationManager>.instance.m_currentFrameIndex;
+            uint simGroup = (uint)transitNodeId >> 7;
+            uint rand = frame - simGroup & 255u;
+            RoadBaseAI.GetTrafficLightState(
+                transitNodeId,
+                ref netMan.m_segments.m_buffer[position.m_segment],
+                frame - simGroup,
+                out RoadBaseAI.TrafficLightState vehicleLightState,
+                out RoadBaseAI.TrafficLightState pedestrianLightState,
+                out bool vehicles,
+                out bool pedestrians);
+
+            if (vehicles || rand < 196u) {
+                return;
+            }
+
+            vehicles = true;
+            RoadBaseAI.SetTrafficLightState(
+                transitNodeId,
+                ref netMan.m_segments.m_buffer[position.m_segment],
+                frame - simGroup,
+                vehicleLightState,
+                pedestrianLightState,
+                true,
+                pedestrians);
         }
 
         [RedirectReverse]
         [MethodImpl(MethodImplOptions.NoInlining)]
-        protected static bool CheckOverlap(ushort vehicleID, ref Vehicle vehicleData, Segment3 segment, ushort ignoreVehicle) {
-            Log.Error("CustomTrainAI.CheckOverlap (1) called.");
+        [UsedImplicitly]
+        protected static bool CheckOverlap(ushort vehicleId,
+                                           ref Vehicle vehicleData,
+                                           Segment3 segment,
+                                           ushort ignoreVehicle) {
+            Log._DebugOnlyError("CustomTrainAI.CheckOverlap (1) called.");
```

Comment:
need to call Error, not _DebugOnlyError

Reply:
in Release the function is not eliminated, the Log call is eliminated